applications/CompressiblePotentialFlowApplication/python_scripts/potential_flow_adjoint_solver.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+# importing the Kratos Library
+import KratosMultiphysics
+import KratosMultiphysics.CompressiblePotentialFlowApplication as KCPFApp
+from KratosMultiphysics.CompressiblePotentialFlowApplication.potential_flow_solver import PotentialFlowSolver
+import KratosMultiphysics.StructuralMechanicsApplication
+
+def CreateSolver(model, custom_settings):
+    return PotentialFlowAdjointSolver(model, custom_settings)
+
+class PotentialFlowAdjointSolver(PotentialFlowSolver):
+    def __init__(self, model, custom_settings):
+
+        self.response_function_settings = custom_settings["response_function_settings"].Clone()
+        self.sensitivity_settings = custom_settings["sensitivity_settings"].Clone()
+        custom_settings.RemoveValue("response_function_settings")
+        custom_settings.RemoveValue("sensitivity_settings")
+        # Construct the base solver.
+        super(PotentialFlowAdjointSolver, self).__init__(model, custom_settings)
+        self.print_on_rank_zero("::[PotentialFlowAdjointSolver]:: ", "Construction finished")
```

Comment:
please use the logger directly, this was only a temp solution until the logger supports MPI (which) it now does
This method is deprecated and will issue deprecation-warnings in the near future